File: Tree/1161. Maximum Level Sum of a Binary Tree.js
var maxLevelSum = function(root) {
    let max = -Infinity;
    const stack = [root];
    let count = 1;
    let current = 0;
    let sum = 0;
    let level = 1;
    let depth = 1;
    while(stack.length) {
        const node = stack.shift();
        if (node) {
            current++;
            sum += node.val;
            if (node.left) {
                stack.push(node.left);
            }
            if (node.right) {
                stack.push(node.right);
            }
            if (current === count) {
                if (sum > max) {
                    max = sum;
                    level = depth;
                }
                current = 0;
                sum = 0;
                count = stack.length;
                depth++;
            }
        }
    }
    return level;
};

File: Golang/Tree/1161.maximum-level-sum-of-a-binary-tree.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func maxLevelSum(root *TreeNode) int {
	max, level := root.Val, 1
	ret := level
	stack := []*TreeNode{root}
	for len(stack) > 0 {
		child := []*TreeNode{}
		temp := 0
		for _, s := range stack {
			if s.Left != nil {
				temp += s.Left.Val
				child = append(child, s.Left)
			}
			if s.Right != nil {
				temp += s.Right.Val
				child = append(child, s.Right)
			}
		}
		level++
		// Edge case if it's the last layer
		if len(child) > 0 {
			if max < temp {
				max = temp
				ret = level
			}
		}

		stack = child
	}
	return ret
}
